lib/api/authentication/api_authentication.py

```diff
@@ -21,12 +31,15 @@ def authenticate(self, request):
         # Unauthenticated, CSRF validation not required
         if not user or not user.is_active:
             return None
-        
+
         self.enforce_csrf(request)
         
         if 'Authorization' in request.headers:
             token = request.headers['Authorization'].split()[1]
-            return (user, token)
+            payload = jwt.decode(token, settings.SECRET_KEY) #TODO check if valid and handle bad signature
+            web_token = WebToken(payload['uid'], payload['course'])
```

Comment:
You could move the decode to the token class:

```python
class WebToken:
    @classmethod
    def from_jwt(cls, token, secret=None):
        if not secret:
            secret = settings.SECRET_KEY
        payload = jwt.decode(token, secret)
        return cls(payload)

    def __init__(self, payload=None):
        self._payload = payload
        self.user_id = payload.get('uid')
        self.course_id = payload.get('course')

# usage:
web_token = WebToken.from_jwt(token)
```

btw. you could also wrap encoding to the class (though, then you should the class to another file)

```python
class WebToken:
    def __init__(self, payload=None):
        self._payload = payload or {}

    @property
    def user_id(self):
        return self._payload.get('uid')

    @user_id.setter
    def user_id(self, value):
        self._payload['uid'] = value

    def _user_set(self, user):
        if user:
            self.user_id = user.id
    user = property(None, _user_set) # getter might not be so relevant

    def encode(self, secret=None):
        if not secret:
            secret = settings.SECRET_KEY
        return jwt.encode(self._payload, secret, 'HS256')

# usage:
wt = WebToken()
wt.user = user
wt.course = course
token = wt.encode()
```

Remember, these are just some ideas for you to think about if you like or not.